example-projects/actix-web-example/src/main.rs

```diff
@@ -39,3 +39,33 @@ async fn main() -> std::io::Result<()> {
     .run()
     .await
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[actix_rt::test]
+    async fn test_post() {
+        use actix_web::{test, App};
+        use cloudevents::AttributesReader;
+
+        let expected = v10::minimal();
+
+        let req = test::TestRequest::post()
+        .header("ce-specversion", "1.0")
+        .header("ce-id", "0001")
+        .header("ce-type", "test_event.test_application")
+        .header("ce-source", "http://localhost/")
+        .to_request();
+
+        let mut app = test::init_service(App::new().service(post_event)).await;
+        let resp = test::call_service(&mut app, req).await;
+        assert!(resp.status().is_success());
```

Comment:
This is exactly the kind of test I _didn't_ want in the example :smile: 

Because `Event` now implements `FromRequest`, we don't need all the HTTP noise in our tests. We can test the handler purely in terms of CloudEvents, like [this](https://github.com/boson-project/func/blob/4dde6cd88a035a86798dbdeb0da72b7ccc8a3f87/templates/rust/events/src/handler.rs#L28-L38)

Reply:
So did I. I tried that first, but for some reason you can't test functions with attributes (`#[post("/")]`) like that without compiler complaining. And I didn't want to change things inside the example before we all agree on what we want to do.

Reply:
Ah, good point! I recall running into that, too. I very much wish that attribute didn't prevent the function from being called. :(